### PR TITLE
Move CLI to commander

### PR DESCRIPTION
**Pull request by frxnz**
I ended up down kind of a weird road on this before coming back to reality. "Interchange" and the CLI itself are in separate files now. This is a result of thinking I needed to do something with commander that ended up being wrong. If you're not happy with this I can revert that portion.

However, I decided to commit it this way because it opens up the possibility of using interchange programmatically in the future. I tested as much as I could and everything seems fine.

### CODE REVIEW

**Comment by frxnz** [comment]
Bonus, check out this lovely `--help`

```bash
notbook at notbook in ~/git/nodebots-interchange on commander
$ interchange -h

  Usage: interchange [options] [command]


  Commands:

    list|l                          Lists all of the available firmwares
    ports|p [options]               Lists all of the attached boards and their ports
    read|r [options]                Read firware info from port
    install|i [options] [firmware]  Test

  Options:

    -h, --help     output usage information
    -V, --version  output the version number


notbook at notbook in ~/git/nodebots-interchange on commander
$ interchange install -h

  Usage: install|i [options] [firmware]

  Test

  Options:

    -h, --help               output usage information
    -a, --board <board>      Type of board/AVR
    -p, --port <port>        Serial port board is attached to
    -f, --firmata            Install firmata version of firmware
    -i, --address <address>  Specify I2C address
```

**Comment by ajfisher** [comment]
Ooh nice one. Will check out your branch shortly and have a look at the code

File: bin/cli.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    .action(interchange.list_devices);
+
+program.command("ports")
+    .description("Lists all of the attached boards and their ports")
+    .alias("p")
+    .option("-v, --verbose", "List with additional information")
+    .action(interchange.list_ports);
+
+program.command("read")
+    .description("Read firware info from port")
```
Comment (ajfisher) [inline bin/cli.js]: typo - "firmware"

File: bin/cli.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        interchange.get_firmware_info(opts.port);
+    });
+
+program.command("install [firmware]")
+    .description("Install specified firmware to board")
+    .alias("i")
+    .option("-a, --board <board>", "Type of board/AVR")
+    .option("-p, --port <port>", "Serial port board is attached to")
+    .option("-f, --firmata", "Install firmata version of firmware")
+    .option("-i, --address <address>", "Specify I2C address")
```
Comment (ajfisher) [inline bin/cli.js]: I didn't have this in my version but we should probably note here that it should be in 0x form so maybe just as an example `eg 0x67`

File: bin/cli.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        interchange.get_firmware_info(opts.port);
+    });
+
+program.command("install [firmware]")
+    .description("Install specified firmware to board")
+    .alias("i")
+    .option("-a, --board <board>", "Type of board/AVR")
+    .option("-p, --port <port>", "Serial port board is attached to")
+    .option("-f, --firmata", "Install firmata version of firmware")
```
Comment (ajfisher) [inline bin/cli.js]: I had forgotten to update this as well but this probably needs to be:

"-f, --firmata=[firmata name]", "Install the firmata version of the firmware, with optional name if more than one are available"

**Comment by ajfisher** [comment]
Looks pretty good other than those little nits which were largely mine anyway ;P

After that's done I'll merge it.

**Pull request merged**